panic in workerNodeTunneler.Setup()

Running the libvirt e2e tests to check https://github.com/confidential-containers/cloud-api-adaptor/pull/1191 , the cloud-api-adaptor bailed out with panic. It turned out that the pod vm had not IP address, so `workerNodeTunneler.Setup()` panic'ed with:

```
panic: runtime error: index out of range [0] with length 0
```

Here is the  cloud-api-adaptor pod's log:

```
$ kubectl logs pod/cloud-api-adaptor-daemonset-fmz86 -n confidential-containers-system
+ exec cloud-api-adaptor libvirt -uri 'qemu+ssh://wmoschet@192.168.122.1/system?no_verify=1' -data-dir /opt/data-dir -pods-dir /run/peerpod/pods -network-name default -pool-name default -socket /run/peerpod/hypervisor.sock
2023/07/17 20:22:31 [adaptor/cloud/libvirt] libvirt config: &libvirt.Config{URI:"qemu+ssh://wmoschet@192.168.122.1/system?no_verify=1", PoolName:"default", NetworkName:"default", DataDir:"/opt/data-dir", VolName:"podvm-base.qcow2"}
cloud-api-adaptor version v0.6.0-dev
  commit: 5997d8183cab6a883e5bd763301ce17425642989
  go: go1.20.6
cloud-api-adaptor: starting Cloud API Adaptor daemon for "libvirt"
2023/07/17 20:22:34 [adaptor] server config: &adaptor.ServerConfig{TLSConfig:(*tlsutil.TLSConfig)(0xc000330280), SocketPath:"/run/peerpod/hypervisor.sock", CriSocketPath:"", PauseImage:"", PodsDir:"/run/peerpod/pods", ForwarderPort:"15150", ProxyTimeout:300000000000}
2023/07/17 20:22:34 [util/k8sops] initialized PeerPodService
2023/07/17 20:22:34 [adaptor] server started
2023/07/17 20:22:35 [podnetwork] routes on netns /var/run/netns/cni-9eb8fbfd-3a1b-6c1c-823f-baf10e4fc5c9
2023/07/17 20:22:35 [podnetwork]     default via 10.244.1.1 dev eth0
2023/07/17 20:22:35 [podnetwork]     10.244.0.0/16 via 10.244.1.1 dev eth0
2023/07/17 20:22:35 [adaptor/cloud] stored /run/peerpod/pods/78e107a6c43233c47b5d6921e11829446624c389ca5f892aa103d93b721d2d38/daemon.json
2023/07/17 20:22:35 [adaptor/cloud] create a sandbox 78e107a6c43233c47b5d6921e11829446624c389ca5f892aa103d93b721d2d38 for pod nginx in namespace default (netns: /var/run/netns/cni-9eb8fbfd-3a1b-6c1c-823f-baf10e4fc5c9)
2023/07/17 20:22:35 [adaptor/cloud/libvirt] Checking if instance (podvm-nginx-78e107a6) exists
2023/07/17 20:22:35 [adaptor/cloud/libvirt] Uploaded volume key /var/lib/libvirt/images/podvm-nginx-78e107a6-root.qcow2
2023/07/17 20:22:35 [adaptor/cloud/libvirt] Create cloudInit iso
Created libvirt connectionExecuting genisoimage
Total translation table size: 0
Total rockridge attributes bytes: 331
Total directory bytes: 0
Path table size(bytes): 10
Max brk space used 0
184 extents written (0 MB)
2023/07/17 20:22:35 [adaptor/cloud/libvirt] Created cloudInit iso
Uploading iso file: /opt/data-dir/podvm-nginx-78e107a6-cloudinit.iso
2023/07/17 20:22:35 [adaptor/cloud/libvirt] 376832 bytes uploaded
2023/07/17 20:22:35 [adaptor/cloud/libvirt] Volume ID: /var/lib/libvirt/images/podvm-nginx-78e107a6-cloudinit.iso
2023/07/17 20:22:35 [adaptor/cloud/libvirt] Create XML for 'podvm-nginx-78e107a6'
2023/07/17 20:22:35 [adaptor/cloud/libvirt] Creating VM 'podvm-nginx-78e107a6'
2023/07/17 20:22:35 [adaptor/cloud/libvirt] Starting VM 'podvm-nginx-78e107a6'
2023/07/17 20:22:35 [adaptor/cloud/libvirt] VM id 29
2023/07/17 20:23:05 [adaptor/cloud/libvirt] domain IP details []
2023/07/17 20:23:05 [adaptor/cloud/libvirt] Instance created successfully
2023/07/17 20:23:05 [adaptor/cloud/libvirt] created an instance podvm-nginx-78e107a6 for sandbox 78e107a6c43233c47b5d6921e11829446624c389ca5f892aa103d93b721d2d38
2023/07/17 20:23:05 [adaptor/cloud] failed to create PeerPod: the server could not find the requested resource (post peerPods.confidentialcontainers.org)
2023/07/17 20:23:05 [adaptor/cloud] created an instance podvm-nginx-78e107a6 for sandbox 78e107a6c43233c47b5d6921e11829446624c389ca5f892aa103d93b721d2d38
panic: runtime error: index out of range [0] with length 0

goroutine 39 [running]:
github.com/confidential-containers/cloud-api-adaptor/pkg/podnetwork/tunneler/vxlan.(*workerNodeTunneler).Setup(0x33f2f4b?, {0xc0000241c0, 0x37}, {0x0?, 0x0?, 0x52fee0?}, 0xc00015a8f0)
        /work/pkg/podnetwork/tunneler/vxlan/workernode.go:40 +0x13af
github.com/confidential-containers/cloud-api-adaptor/pkg/podnetwork.(*workerNode).Setup(0xc000385630?, {0xc0000241c0, 0x37}, {0x0, 0x0, 0x0}, 0xc00015a8f0)
        /work/pkg/podnetwork/workernode.go:189 +0xd6
github.com/confidential-containers/cloud-api-adaptor/pkg/adaptor/cloud.(*cloudService).StartVM(0xc0004ce000, {0x3890988, 0xc0005221b0}, 0x10?)
        /work/pkg/adaptor/cloud/cloud.go:305 +0x5e4
github.com/kata-containers/kata-containers/src/runtime/protocols/hypervisor.RegisterHypervisorService.func2({0x3890988, 0xc0005221b0}, 0xc0004442c0)
        /go/pkg/mod/github.com/kata-containers/kata-containers/src/runtime@v0.0.0-20230606170044-338e18e4fd46/protocols/hypervisor/hypervisor.pb.go:946 +0x98
github.com/containerd/ttrpc.defaultServerInterceptor({0x3890988?, 0xc0005221b0?}, 0x15?, 0xc000142560?, 0x7?)
        /go/pkg/mod/github.com/containerd/ttrpc@v1.1.0/interceptor.go:45 +0x26
github.com/containerd/ttrpc.(*serviceSet).dispatch(0xc00044cd60, {0x3890988, 0xc0005221b0}, {0xc00053c000, 0x15}, {0xc0000f8014, 0x7}, {0xc0001c4000, 0x42, 0x50})
        /go/pkg/mod/github.com/containerd/ttrpc@v1.1.0/services.go:95 +0x1be
github.com/containerd/ttrpc.(*serviceSet).call(0xc000316080?, {0x3890988?, 0xc0005221b0?}, {0xc00053c000?, 0x0?}, {0xc0000f8014?, 0xc0005241e0?}, {0xc0001c4000, 0x42, 0x50})
        /go/pkg/mod/github.com/containerd/ttrpc@v1.1.0/services.go:64 +0x71
github.com/containerd/ttrpc.(*serverConn).run.func2(0x1)
        /go/pkg/mod/github.com/containerd/ttrpc@v1.1.0/server.go:438 +0xe5
created by github.com/containerd/ttrpc.(*serverConn).run
        /go/pkg/mod/github.com/containerd/ttrpc@v1.1.0/server.go:434 +0x7b1
```

--- a/pkg/podnetwork/tunneler/vxlan/workernode.go
+++ b/pkg/podnetwork/tunneler/vxlan/workernode.go
@@ -34,7 +34,15 @@ func (t *workerNodeTunneler) Setup(nsPath string, podNodeIPs []netip.Addr, confi
 
 	var dstAddr netip.Addr
 
+	numIPs := len(podNodeIPs)
+	if numIPs == 0 {
+		return fmt.Errorf("pod node has no IPs")
+	}
+
 	if config.Dedicated {
+		if numIPs < 2 {
+			return fmt.Errorf("dedicated tunnel missing destination address")
+		}
 		dstAddr = podNodeIPs[1]
 	} else {
 		dstAddr = podNodeIPs[0]

